src/Components/AppShell/AppShell.js

```diff
@@ -108,25 +108,45 @@ class AppShell extends Component {
 
   onCloseFatalErrorModal = () => {}
 
+  // grabs the last 500 log lines as a string formatted for inclusion as a github issue
+  prepareLogLines () {
+    if (this.props.logs.lines) {
+      let firstLogTime = this.props.logs.lines[0].time.getTime()
+      return this.props.logs.lines
+      .slice(-500) 
+      .map(v => `T+${v.time.getTime() - firstLogTime}ms: ${v.line}`)
+        .join('\n')
+    }
+
+    return ''
+  }
+
+  // Remove any user-specific paths in exception messages
+  sanitizePaths(message) {
+    // Prepare our paths so we *always* will get a match no matter
+    // path separator (oddly, on Windows, different errors will give
+    // us different path separators)
+    var appPath = app.getAppPath().replace(/\\/g, "/")
+
+    // I couldn't figure out the regex, so a loop will do.
+    while (message.indexOf(appPath) >= 0) {
+      message = systemError.replace(appPath, "")
+    }
+
+    return message;
+  }
+
   render () {
     const path = this.props.location.pathname
     const segment = path.replace(/^\//g, '').replace(/\//g, '-') || 'root'
     let systemError = this.props.core.systemError
-
+    let logLines = ''
     if (systemError) {
       systemError = systemError.stack || systemError
```

Comment:
The fact that we have this branch kind of makes me feel like we should pull the `<Modal>` out into its own component and just reference that.